### PR TITLE
ramkrishna2910/bug fixes

### LINKED ISSUE #224: Benchmarking docker not stopped if a run fails
```
mlagility.api.devices.BenchmarkException: Error: Failure to run model using ORT - ERROR:root:Subprocess failed with command: docker run -d --rm -v tmp_cache/sample/x86_benchmark:/app --name mlagility-onnxruntime-mlas-ep mlagility-onnxruntime-image and error message: docker: Error response from daemon: Conflict. The container name "/mlagility-onnxruntime-mlas-ep" is already in use by container "5a5e5eb76a6069b3a8c600d9d597951c8673db404c87b90ba82d51ba1b8a9a7d". You have to remove (or rename) that container to be able to reuse that name.
```
Container needs to be removed in any case so the container name does not conflict with the next run.

### PR DESCRIPTION
**Pull request by ramkrishna2910**
This MR fixes the following bug

-  Ensures the container is stopped and removed even when the benchmarking run fails
- Handles custom cache dir for benchmarking runs
- Resolves a bug in choosing the right input type for ort CPU execution

closes #224

### CODE REVIEW

File: src/mlagility/api/run_ort_model.py [hunk 1]
```diff
@@ -55,11 +55,11 @@ def _dummy_inputs(sess_input) -> dict:
...
         datatype = "float16"
-    if dtype_str == "float":
+    elif dtype_str == "float":
         datatype = "float32"
-    if dtype_str == "double":
+    elif dtype_str == "double":
         datatype = "float64"
-    if dtype_str == "long":
+    elif dtype_str == "long":
         datatype = "int64"
```
Comment (ramkrishna2910) [inline src/mlagility/api/run_ort_model.py:63]: Lets just act like this never happened😅

File: src/mlagility/api/devices.py [hunk 2]
```diff
@@ -386,6 +386,7 @@ def execute_trt_remotely(
     _ip, username = configure_remote(device)
```
Comment (jeremyfowers) [inline src/mlagility/api/devices.py:386]: increment patch version number of mlagility so we can release to PyPI when this merges :)

File: src/mlagility/api/run_ort_model.py (hunk 1 above)
Comment (jeremyfowers) [reply src/mlagility/api/run_ort_model.py:63]: deal!

**Review by jeremyfowers — APPROVED** [review]
lgtm!

**Pull request merged**